rlbot_gui/gui/js/botmanager.js

```diff
@@ -0,0 +1,154 @@
+					globalJson.repos[index].gamemodes = myJson.gamemodes;
+					globalJson = json;
+				});
+		}
+
+		displayRepoData()
+
+	}
+
+	async function displayRepoData(){
+		document.getElementById('main').innerHTML = '';
+		//make bot ui elements out of json data
+		for (var i = 0; i < globalJson.repos.length; i++) {
+			repo = globalJson.repos[i];
+			display = true;
+
+			//fill gamemode filters
+			var gamemodeFilters = [];
+			for (var j = document.getElementsByName('gamemode').length - 1; j >= 0; j--) {
+				if(document.getElementsByName('gamemode')[j].checked){
+					arrLen = gamemodeFilters.length;
+					gamemodeFilters[arrLen] = document.getElementsByName('gamemode')[j].value;
+				}
+			}
+
+			var gamemodeNotFoundCount = 0;
+			if (gamemodeFilters.length != 0) {
+				//if any active gamemode gamemodeFilters, apply them
+				var gamemodeNotFoundCount = gamemodeFilters.length
+				for (var gamemodeFilterIndex = gamemodeFilters.length - 1; gamemodeFilterIndex >= 0; gamemodeFilterIndex--) {
+					for (var j = repo.gamemodes.length - 1; j >= 0; j--) {
+						if (repo.gamemodes[j].name.toLowerCase() == gamemodeFilters[gamemodeFilterIndex].toLowerCase()) {
+							--gamemodeNotFoundCount;
+						}
+					}
+				}
+			}
+			if (gamemodeNotFoundCount != 0 || repo.name.toLowerCase().indexOf(document.getElementById('searchBotName').value.toLowerCase())==-1) {
+				display = false
+			}
+
+			if (display) {
+
+				gamemodeString = '';
+				for (var index = repo.gamemodes.length - 1; index >= 0; index--) {
+					gamemodeString+=repo.gamemodes[index].name;
+					if (index!=0) {
+						gamemodeString += ', ';
+					}
+				}
+
+				categoryString = '';
+				for (var index = repo.categories.length - 1; index >= 0; index--) {
+					categoryString+=repo.categories[index].name;
+					if (index!=0) {
+						categoryString += ', ';
+					}
+				}
+
+
+
+				//Filling in template and inserting it
+				var html = '';
+
+				html +=	"<div class=\"md-card settings-card md-theme-default\">";
+				html +=		"<div class=\"md-card-header\">";
+	        	html += 		"<div class=\"md-title\">"+repo.name+"</div>";
+	        	html +=		"</div>";
+	        	html +=		"<div class=\"md-card-content\">";
+				html += 		"<div class=\"center-flex\">";
+				html += 			"<div class=\"md-layout md-gutter\">";
+				html += 			    "<div class=\"md-layout-item\"><label for=\"gamemodes-0\">gamemode</label>";
+				html += 			        "<p id=\"gamemodes-0\" class=\"managerInfo\">"+gamemodeString+"</p>";
+				html += 			    "</div>";
+				html += 			    "<div class=\"md-layout-item\"><label for=\"type-0\">Category</label>";
+				html += 			        "<p id=\"type-0\" class=\"managerInfo\">"+categoryString+"</p>";
+				html += 			    "</div>";
+				html += 			"</div> ";
+				html += 			"<span style=\"flex-grow: 1;\"></span> ";
+				html += 			"<button type=\"button\" class=\"md-button md-primary md-raised md-theme-default\">";
+				html += 				"<div id=\"button"+i+"\" class=\"md-ripple\">";
```

Comment:
you can wrap the outside with single quotes then you don't need the backslash.

Reply:
also instead of creating html this way it may be better to use dom element creation.

This way can lead to XSS much easier.
You could then create most of the html file and import it using html imports / html templates.  And then just fill in the rest with dom element creation.

Reply:
single quotes are used for js function parameters and the escaped double quote for the html property value so the double quote can be used for the string

Reply:
We're using a client side templating library called Vue, I suspect we can leverage it to clean up this code a lot. If you don't have time, you can give me write permissions on your repo and I'll make a commit on this PR.

Reply:
Yeah, that should make the code more in line with the project. If you want to do that it would be great (altrough im the type of person who prefers not to use templating engines)